File: Data Structures & Algorithms/climbing-stairs/submission-1.java
class Solution {
    public int climbStairs(int n) {
        int result = 0;
        int n1 = 0;
        int n2 = 1;
        for (int i = 0;i < n;i++) {
            result = n1 + n2;
            n1 = n2;
            n2 = result;
        }
        return result;
    }
}

File: Data Structures & Algorithms/k-closest-points-to-origin/submission-1.kt
class Solution {
    fun kClosest(points: Array<IntArray>, k: Int): Array<IntArray> {
        val queue = PriorityQueue<IntArray>() { a, b ->
        (a[0] * a[0] + a[1] * a[1]) - (b[0] * b[0] + b[1] * b[1])
        }
        for(point in points) queue.add(point)
        val result = mutableListOf<IntArray>()
        repeat(k) {
            result.add(queue.poll())
        }
        return result.toTypedArray()
    }
}
